src/openms/source/QC/TopNoverRT.cpp

```diff
@@ -0,0 +1,168 @@
+	// MS2-Spektren without mate are added in unassignedPeptideIdentifications (only Information m/z and RT)
+	void TopNoverRT::compute(const MSExperiment& exp, FeatureMap& features)
+	{
+		if (features.empty())
+		{
+			LOG_WARN << "The FeatureMap is empty.\n";
+			features.setUnassignedPeptideIdentifications({});
+		}
+		if (exp.empty())
+		{
+			throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The mzml file / MSExperiment is empty.\n");
+		}
+		ms2_included_.clear();
+		//ms2_included_.resize(exp.getSpectra.size(),make_pair(0,nullptr));
+		setScanEventNumber_(exp);
+		//if MS2-spectrum PeptideIdentifications found ->  ms2_included_ nullptr to PepID pointer
+		setPresenceAndScanEventNumber_(exp, features);
+		//if Ms2-spectrum not identified, add to unassigned PeptideIdentification without ID, contains only RT and ScanEventNumber
+		addUnassignedPeptideIdentification_(exp, features);
+	}
+	//if ms2 spetrum not included, add to unassignedPeptideIdentification, set m/z and RT values
+	void TopNoverRT::setScanEventNumber_(const MSExperiment& exp)
+	{
+		UInt32 scan_event_number{ 0 };
+		for (MSSpectrum spec : exp.getSpectra())
+		{
+			if (spec.getMSLevel() == 1)
+			{
+				scan_event_number = 0;
+				ms2_included_.push_back(make_pair(scan_event_number, false));
+			}
+			else if (spec.getMSLevel() == 2)
+			{
+				++scan_event_number;
+				ms2_included_.push_back(make_pair(scan_event_number, false));
+			}
+		}
+	}
+
+	void TopNoverRT::setPresenceAndScanEventNumber_(const MSExperiment& exp, FeatureMap& features )
+	{
+		for (Feature& feature : features)
+		{
+			for (PeptideIdentification& peptide_ID : feature.getPeptideIdentifications())
+			{
+				MSExperiment::ConstIterator it = exp.RTBegin(peptide_ID.getRT() - EPSILON_);
+				if (it == exp.end())
+				{
+					throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The retention time of the MZML and featureXML file does not match.");
+				}
+
+				const auto& spectrum = *it;
+				if (spectrum.getRT() - peptide_ID.getRT() > EPSILON_)
+				{
+					throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "PeptideID with RT " + to_string(peptide_ID.getRT()) + " s does not have a matching MS2 spectrum. Closest RT was " + to_string(spectrum.getRT()) + ", which seems too far off.\n");
+				}
+
+				if (spectrum.getMSLevel() == 2)
+				{
+					ms2_included_[distance(exp.begin(), it)].second = true;
+					peptide_ID.setMetaValue("ScanEventNumber", ms2_included_[distance(exp.begin(), it)].first);
+					peptide_ID.setMetaValue("identified", '+');
+				}
+				else
+				{
+					throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The matching retention time of the MZML has the wrong MSLevel");
+				}	
+			}
+		}
+		//marks all seen unassignedPeptideIdentifications in vector ms2_included
+		for (PeptideIdentification& unassigned_ID : features.getUnassignedPeptideIdentifications())
```

Comment:
Code is identical to PepIDs for Features, right? refactor into a separate function

Reply:
..and avoid code duplication